Site/PullRequestPreviewBranchWorkflow.md

```diff
@@ -0,0 +1,132 @@
+
+This is a simplification of the [throw-away integration test
+branch](https://docs.google.com/document/d/1uVQYI2cmNx09fDkHDA136yqDTqayhxqfvjFiuUue7wo#heading=h.2r0g9kvx5b2a)
+where the `preview` branch takes the place of the 'next' branch.
+
+In this setup, the `production bssw.io` site is generated from the `master`
+branch while the `preview.bssw.io` site is generated from the `preview`
+branch.
+
+Some simple rules for the workflow:
+
+* To preview the content, merge the `content-X` branch to the `preview` branch
+  and observe it on the `preview.bssw.io` site.
+
+* All edits and reviews to the proposed content are performed on the
+  `content-X` branch and the associated PR before merging to the `master`
+  branch.
+
+* When the proposed content is 100% ready to publish, the `content-X` branch
+  is "graduated" (i.e. is merged to the `master` branch by merging to
+  associated PR).
+
+* The `preview` branch always contains `master` so every merge to `master`
+  must be followed up by a merge of `master` to `preview`.
+
+* The `preview` branch is **never** merged to the `master` branch.  (The
+  `preview` branch is disposable and can be rebuilt from scratch at any time.)
+
+
+## Proposed Implementation
+
+The merge of the `content-X` branch is performed by GitHub by merging the
+associated PR.  Pushing new commits to the `content-X` branch are performed as
+usual.  Because the PR has been created, every GitHub user with push access to
+the BSSw GitHub repo can push commits to the authors '<author-id>/content-x'
+branch.  Therefore, the only operations that are not already coverted are the
+merges to the `preview` branch from each PR branch and `master`.
+
+The remaining operations of the proposed workflow can be performed manually on
+the shell command-line, or using GitHub PRs on the web interface, or could be
+automated with GitHub Actions or Travis CI jobs.  It would be highly desirable
+to automate these merges to the `preview` branch as the workflow would have
+near-zero manual overhead as compared to the current workflow (where the
+`content-X` branch is merged to `master` very early and subsequent edits are
+done un-reviewed on `master`).
+
+The commands to merge an updated `content-X` branch to `preview` are:
+
+```
+[(<branch>)]$ git checkout preview
+[ (preview)]$ git pulll       # from origin/preview
+[ (preview)]$ git remote add <author-id> git@github.com:<author-id>/betterscientificsoftware.io.git
+[ (preview)]$ git fetch <author-id>
+[ (preview)]$ git merge <author-id>/content-x
+[ (preview)]$ git push       # to origin/preview
+```
+
+The commands to update the `preview` branch after the `content-X` branch and
+PR has been "graduated" (i.e. merged to `master`) are:
+
+```
+[(<branch>)]$ git checkout preview
+[ (preview)]$ git pulll      # from origin/preview
+[ (preview)]$ git merge origin/master
+[ (preview)]$ git push       # to origin/preview
+```
```

Comment:
> Are the `[(<branch>)]$` and `[(preview)]$` meant to denote the working directory in which these commands are issued?

No this is the current branch as displayed by the [git shell prompt script](https://github.com/git/git/blob/master/contrib/completion/git-prompt.sh).   I just left out the working directory (because it is supper long for this github repo).